src/appleseed/foundation/utility/iesparser.cpp

```diff
@@ -415,466 +403,470 @@ void IESParser::parse_tilt_data(std::istream& input_stream)
+                    "With photometric type C, the last horizontal angle "
+                    "can be only 0, 90, 180 or 360 degrees", m_line_counter);
             }
         }
         else
         {
-            throw ParsingException("With photometric type C, the first horizontal angle"
-                " can be only 0 or 90 degrees", line_counter);
+            throw ParsingException(
+                "With photometric type C, the first horizontal angle "
+                "can be only 0 or 90 degrees", m_line_counter);
         }
     }
-    else if (photometric_type == PhotometricTypeA ||
-        photometric_type == PhotometricTypeB)
+    else if (m_photometric_type == PhotometricTypeA ||
+             m_photometric_type == PhotometricTypeB)
     {
-        if (feq(horizontal_angles.front(), 0.0) &&
-            feq(horizontal_angles.back(), 90.0))
+        if (feq(m_horizontal_angles.front(), 0.0) &&
+            feq(m_horizontal_angles.back(), 90.0))
         {
-            symmetry = SymmetricHalvesX;
+            m_symmetry = SymmetricHalvesX;
         }
-        else if (feq(horizontal_angles.front(), -90.0) &&
-            feq(horizontal_angles.back(), 90.0))
+        else if (feq(m_horizontal_angles.front(), -90.0) &&
+                 feq(m_horizontal_angles.back(), 90.0))
         {
-            symmetry = NoSymmetry;
+            m_symmetry = NoSymmetry;
         }
         else
         {
-            throw ParsingException("With photometric type C, the first horizontal angle"
-                " can be only 0 or -90 degrees, the last horizontal angle should be 90 degrees",
-                line_counter);
+            throw ParsingException(
+                "With photometric types A and B, the first horizontal angle "
+                "can be only 0 or -90 degrees, and the last horizontal "
+                "angle must be 90 degrees", m_line_counter);
         }
     }
-    if (!check_increasing_order(horizontal_angles))
+    if (!check_increasing_order(m_horizontal_angles))
     {
-        throw ParsingException("Horizontal angles"
-            " must be mentioned in increasing order", line_counter);
+        throw ParsingException(
+            "Horizontal angles must be mentioned in increasing order", m_line_counter);
     }
 }
 
-
 bool IESParser::is_keyword_line(const std::string& line)
 {
-    boost::regex regex(KEYWORD_LINE_REGEX);
-    return boost::regex_match(line, regex);
+    static const boost::regex Regex(KeywordLineRegex);
+    return boost::regex_match(line, Regex);
 }
 
-
 bool IESParser::is_tilt_line(const std::string& line)
 {
-    boost::regex regex(TILT_LINE_REGEX);
-    return boost::regex_match(line, regex);
+    static const boost::regex Regex(TiltLineRegex);
+    return boost::regex_match(line, Regex);
 }
 
-
 void IESParser::parse_keyword_line(const std::string& line)
 {
-    boost::regex regex(KEYWORD_LINE_REGEX);
+    static const boost::regex Regex(KeywordLineRegex);
```

Comment:
This is subtly dangerous (I think): by marking this variable static, the compiler will ensure that there is a single instance of it shared across all calls, and that it is constructed "some time" before the method is first invoked. The trouble is, there isn't any guarantee that this "lazy construction" will work properly if the method is invoked by multiple threads at the same time. The chance of this failing is rather tiny, but as far as I know, it exists. I would suggest to simply mark the variable as `const`, and remove `static`.

On a sidenote, even though the variable is const, it's not a true constant fully defined at compile-time (e.g. `const int Something = 12`) but rather just a normal variable that's made immutable. As a normal variable, we would spell it in lower case: `regex` (or `re` if you need to avoid conflicts with `boost::regex`).